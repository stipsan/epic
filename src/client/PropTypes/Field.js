import { PropTypes } from 'react'

export default {
  input: PropTypes.shape({
    checked: PropTypes.bool,
    name: PropTypes.string.isRequired,
    onBlur: PropTypes.func.isRequired,
    onChange: PropTypes.func.isRequired,
    onDragStart: PropTypes.func.isRequired,
    onDrop: PropTypes.func.isRequired,
    onFocus: PropTypes.func.isRequired,
    value: PropTypes.any.isRequired,
  }).isRequired,
  meta: PropTypes.shape({
    active: PropTypes.bool,
    asyncValidating: PropTypes.bool.isRequired,
    dirty: PropTypes.bool.isRequired,
    dispatch: PropTypes.func,
    error: PropTypes.string,
    invalid: PropTypes.bool.isRequired,
    pristine: PropTypes.bool.isRequired,
    submitting: PropTypes.bool.isRequired,
    touched: PropTypes.bool.isRequired,
    valid: PropTypes.bool.isRequired,
    visited: PropTypes.bool,
  }).isRequired
}
