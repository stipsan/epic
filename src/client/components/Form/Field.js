import { Field } from 'redux-form/immutable'
export default Field
