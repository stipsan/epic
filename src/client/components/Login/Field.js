import { PropTypes } from 'react'

import cx from './style.scss'

const Field = ({
  input,
  meta,
  ...custom,
  asyncValidating,
  touched,
  error,
  submitting,
  className,
 ...props
}) => (
  <p
    className={cx('control', `control-${custom.name}`, className, {
      error: meta.touched && meta.error,
      'is-loading': meta.asyncValidating,
    })}
  >
    <input
      {...input}
      className={cx(`input-${input.props.name}`, {
        'is-danger': touched && error,
      })}
      autoComplete="email"
      disabled={meta.submitting}
    />
    {meta.touched && meta.error &&
      <span className={cx('help', 'is-danger')}>{meta.error}</span>
    }
  </p>
)

Field.propTypes = {
  asyncValidating: PropTypes.bool,
  className: PropTypes.string,
  error: PropTypes.string,
  name: PropTypes.string.isRequired,
  placeholder: PropTypes.string.isRequired,
  submitting: PropTypes.bool,
  touched: PropTypes.bool,
  type: PropTypes.string,
}

export default Field
