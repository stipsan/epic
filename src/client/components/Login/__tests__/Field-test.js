import renderer from 'react-test-renderer'

import Field from '../Field'

jest.mock('react-collapse', () => 'Collapse')

describe('<Login.Field />', () => {
  const defaultProps = {
    input: {
      name: 'email',
      onBlur: () => {},
      onChange: () => {},
      onDragStart: () => {},
      onDrop: () => {},
      onFocus: () => {},
      value: '',
    },
    meta: {
      asyncValidating: false,
      dirty: false,
      invalid: true,
      pristine: true,
      submitting: false,
      touched: false,
      valid: false,
    }
  }
  it('should render correctly', () => {
    let component = renderer.create(
      <Field {...defaultProps} />
    )
    expect(component.toJSON()).toMatchSnapshot()

    component = renderer.create(
      <Field {...defaultProps} type="email" />
    )
    expect(component.toJSON()).toMatchSnapshot()

    component = renderer.create(
      <Field
        {...defaultProps}
        meta={{
          ...defaultProps.meta,
          touched: true,
          valid: false,
          invalid: true,
          error: 'Required'
        }}
      />
    )
    expect(component.toJSON()).toMatchSnapshot()

    component = renderer.create(
      <Field
        {...defaultProps}
        input={{
          ...defaultProps.input,
          value: 'tony@stark.enterprise'
        }}
        meta={{
          ...defaultProps.meta,
          pristine: false,
          touched: true,
          valid: true,
          invalid: false,
          dirty: true,
          submitting: true
        }}
      />
    )
    expect(component.toJSON()).toMatchSnapshot()

    component = renderer.create(
      <Field
        {...defaultProps}
        input={{ ...defaultProps.input, name: 'password' }}
        type="password"
        icon="lock"
        required
      />
    )
    expect(component.toJSON()).toMatchSnapshot()

    component = renderer.create(
      <Field
        {...defaultProps}
        input={{ ...defaultProps.input, name: 'username' }}
        type="text"
        autoComplete="name"
      />
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
})
