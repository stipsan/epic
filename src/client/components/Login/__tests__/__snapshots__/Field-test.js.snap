exports[`<Login.Field /> should render correctly 1`] = `
<div
  className="uk-form-row">
  <div
    className="uk-form-icon uk-width-1-1">
    <i
      className="uk-icon-email" />
    <input
      autoComplete="email"
      className="uk-width-1-1 uk-form-large"
      disabled={false}
      name="email"
      onBlur={[Function onBlur]}
      onChange={[Function onChange]}
      onDragStart={[Function onDragStart]}
      onDrop={[Function onDrop]}
      onFocus={[Function onFocus]}
      value="" />
  </div>
  <Collapse
    className="uk-form-help-block uk-text-left"
    isOpened={false}>
    &nbsp;
  </Collapse>
</div>
`;

exports[`<Login.Field /> should render correctly 2`] = `
<div
  className="uk-form-row">
  <div
    className="uk-form-icon uk-width-1-1">
    <i
      className="uk-icon-email" />
    <input
      autoComplete="email"
      className="uk-width-1-1 uk-form-large"
      disabled={false}
      name="email"
      onBlur={[Function onBlur]}
      onChange={[Function onChange]}
      onDragStart={[Function onDragStart]}
      onDrop={[Function onDrop]}
      onFocus={[Function onFocus]}
      type="email"
      value="" />
  </div>
  <Collapse
    className="uk-form-help-block uk-text-left"
    isOpened={false}>
    &nbsp;
  </Collapse>
</div>
`;

exports[`<Login.Field /> should render correctly 3`] = `
<div
  className="uk-form-row">
  <div
    className="uk-form-icon uk-width-1-1">
    <i
      className="uk-icon-email" />
    <input
      autoComplete="email"
      className="uk-width-1-1 uk-form-large uk-form-danger"
      disabled={false}
      name="email"
      onBlur={[Function onBlur]}
      onChange={[Function onChange]}
      onDragStart={[Function onDragStart]}
      onDrop={[Function onDrop]}
      onFocus={[Function onFocus]}
      value="" />
  </div>
  <Collapse
    className="uk-form-help-block uk-text-left"
    isOpened={true}>
    Required
  </Collapse>
</div>
`;

exports[`<Login.Field /> should render correctly 4`] = `
<div
  className="uk-form-row">
  <div
    className="uk-form-icon uk-width-1-1">
    <i
      className="uk-icon-email" />
    <input
      autoComplete="email"
      className="uk-width-1-1 uk-form-large"
      disabled={true}
      name="email"
      onBlur={[Function onBlur]}
      onChange={[Function onChange]}
      onDragStart={[Function onDragStart]}
      onDrop={[Function onDrop]}
      onFocus={[Function onFocus]}
      value="tony@stark.enterprise" />
  </div>
  <Collapse
    className="uk-form-help-block uk-text-left"
    isOpened={false}>
    &nbsp;
  </Collapse>
</div>
`;

exports[`<Login.Field /> should render correctly 5`] = `
<div
  className="uk-form-row">
  <div
    className="uk-form-icon uk-width-1-1">
    <i
      className="uk-icon-spinner uk-icon-spin" />
    <input
      autoComplete="email"
      className="uk-width-1-1 uk-form-large"
      disabled={false}
      name="email"
      onBlur={[Function onBlur]}
      onChange={[Function onChange]}
      onDragStart={[Function onDragStart]}
      onDrop={[Function onDrop]}
      onFocus={[Function onFocus]}
      value="" />
  </div>
  <Collapse
    className="uk-form-help-block uk-text-left"
    isOpened={false}>
    &nbsp;
  </Collapse>
</div>
`;

exports[`<Login.Field /> should render correctly 6`] = `
<div
  className="uk-form-row">
  <div
    className="uk-form-icon uk-width-1-1">
    <i
      className="uk-icon-lock" />
    <input
      autoComplete="password"
      className="uk-width-1-1 uk-form-large"
      disabled={false}
      name="password"
      onBlur={[Function onBlur]}
      onChange={[Function onChange]}
      onDragStart={[Function onDragStart]}
      onDrop={[Function onDrop]}
      onFocus={[Function onFocus]}
      required={true}
      type="password"
      value="" />
  </div>
  <Collapse
    className="uk-form-help-block uk-text-left"
    isOpened={false}>
    &nbsp;
  </Collapse>
</div>
`;

exports[`<Login.Field /> should render correctly 7`] = `
<div
  className="uk-form-row">
  <div
    className="uk-form-icon uk-width-1-1">
    <i
      className="uk-icon-username" />
    <input
      autoComplete="name"
      className="uk-width-1-1 uk-form-large"
      disabled={false}
      name="username"
      onBlur={[Function onBlur]}
      onChange={[Function onChange]}
      onDragStart={[Function onDragStart]}
      onDrop={[Function onDrop]}
      onFocus={[Function onFocus]}
      type="text"
      value="" />
  </div>
  <Collapse
    className="uk-form-help-block uk-text-left"
    isOpened={false}>
    &nbsp;
  </Collapse>
</div>
`;
